crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1303,31 +1303,34 @@ impl<'src> Parser<'src> {
             return Expr::Dict(self.parse_dictionary_expression(None, value.expr, start));
         }
 
-        let key_or_value = self.parse_named_expression_or_higher();
+        // For dictionary expressions, the key uses the `expression` rule while for
+        // set expressions, the element uses the `star_expression` rule. So, use the
+        // one that is more general and limit it later.
```

Comment:
Can you point out where "limit it later" is happening, or is this part of another PR (with tests)?

Reply:
Linking for posterity: https://github.com/astral-sh/ruff/pull/10556/files#diff-b0faf8ec2615ebd31aff987f04757ead6fcfef23e995cd370e5da1700e9ae684